### DISCUSSION TITLE
classes/class_callable

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#225] [User-contributed notes]
# classes/class_callable

A built-in type representing a method or a standalone function. Description: Callable is a built-in Variant type that represents a function. It can either be a method within an Object instance, or ...

https://docs.godotengine.org/en/stable/classes/class_callable.html

**Comment by ilexl** [comment]
To create a Callable with parameters in C# the only option is to use lamda. 

Example:
Callable.From((varType varName) => FunctionNameWithParameters(varName))

Source [Godot Issue #68752](https://github.com/godotengine/godot/issues/68752#issuecomment-1317848339)

**Comment by AllMindControl** [comment]
First gdscript example after description uses 
prints(args) 
Instead of 
print(args)

**Reply by Calinou** [reply]
This is not a typo – [`prints()`](https://docs.godotengine.org/en/latest/classes/class_@globalscope.html#class-globalscope-method-prints) exists and prints each parameter separated by a space.

**Comment by cyberlove777** [comment]
C++ Example with arguments (for now v4.4 works only with class methods):
```cpp
//Method to call
void PlayerCPP::SetForceCanJump(bool Can)
{
	ForceCanJump = Can;
}

//..create callable somewhere in PlayerCPP class
Callable C(callable_mp(this, &PlayerCPP::SetForceCanJump));
Array A; A.append(true);
C = C.callv(A);

//Use C as you need, for example with tween
Ref<Tween> XTween = get_tree()->create_tween();
XTween->tween_callback(C);
```

**Reply by cyberlove777** [reply]
Shorter example with C++ and arguments:

Ref<Tween> XTween = get_tree()->create_tween();
XTween->tween_callback((callable_mp(this, &PlayerCPP::SetForceCanJump)).bind(true))->set_delay(1.0f);

**Comment by TedKaczynskisCousin** [comment]
I think the description for `call_deferred()` should mention that there are actually multiple periods of idle time within a frame. For example, a user like myself wanting to call some code at the end of the next frame may do something like this:
```GDscript
func _process(_delta: float) -> void:
	get_tree().process_frame.connect(callable, CONNECT_DEFERRED | CONNECT_ONE_SHOT)
```
However, as it turns out the callable will actually execute at the start of the next frame, much to my bewilderment.
